### ISSUE TITLE
Implicit lval conversions

### ISSUE THREAD

**Issue by jyn514** [jyn514/rcc#26]
```
char *s = "hi";
<stdin>:1:11: error: cannot implicitly convert 'char[3]' to '*char'. help: use an explicit cast: (*char)
```

The following degrade to pointers when used in an expression context:

- Arrays
- Functions

Additionally, ids are automatically derefence (this is implemented).

Waiting on https://github.com/jyn514/rust-c-compiler/issues/19 since this is only an issue for degrading to pointers.

**Comment by jyn514** [comment]
This is _mostly_ done with the exception that `char *s = "some string";` doesn't work. Current behavior: `<stdin>:1:11: error: cannot implicitly convert 'char[12]' to '*char'. help: use an explicit cast: (*char)`.

Now, the _immediate_ fix for this is easy, just call `assignment_expr().rval()` instead of `assignment_expr().rval()` in `Parser::initializer`. However, if we do that, we get a bunch of segfaults in the compiled code because arrays have one fewer level of indirection than pointers, and my code doesn't account for that. I'm leaving this open, but since anyone can use "char[]" instead of "char *" (at least until I get around to implementing function parameters properly), I'm marking it as low priority.

**Comment by jyn514** [comment]
function pointers too:

```c
int f();
int main() {
        int (*fp)() = f;
        return fp();
}
<stdin>:3:24: error: cannot implicitly convert 'int()' to 'int(*)()'. help: use an explicit cast: (int(*)())
```
